### PR TITLE
Reduce counter visibility

### PR DESCRIPTION
**Pull request by janekdb**

### CODE REVIEW

File: community/cypher/spec-suite-tools/src/test/scala/cypher/features/ScenarioTestHelper.scala [hunk 1]
```diff
@@ -50,7 +50,7 @@ object ScenarioTestHelper {
                   debugOutput: Boolean = false): util.Collection[DynamicTest] = {
     val blacklist = config.blacklist.map(parseBlacklist).getOrElse(Set.empty[BlacklistEntry])
     checkForDuplicates(scenarios, blacklist.toList)
-    val (expectFail, expectPass) = scenarios.partition { s => blacklist.exists(_.isBlacklisted(s)) }
+    val (expectFail, expectPass) = scenarios.partition(s => blacklist.exists(_.isBlacklisted(s)))
```
Comment (janekdb) [inline community/cypher/spec-suite-tools/src/test/scala/cypher/features/ScenarioTestHelper.scala:53]: Partial function not required.

File: community/cypher/spec-suite-tools/src/test/scala/cypher/features/ScenarioTestHelper.scala [hunk 2]
```diff
@@ -41,7 +41,7 @@ import scala.util.Success
 import scala.util.Try
 
 object ScenarioTestHelper {
-  var unexpectedSuccessCount = new AtomicInteger(0)
+  private val unexpectedSuccessCount = new AtomicInteger(0)
```
Comment (janekdb) [inline community/cypher/spec-suite-tools/src/test/scala/cypher/features/ScenarioTestHelper.scala:44]: The reference is not accessed outside of this object.

Interestingly the value of the atomic integer is not read so maybe this is here to support debugging.

Comment (klaren) [reply community/cypher/spec-suite-tools/src/test/scala/cypher/features/ScenarioTestHelper.scala:44]: The CI builds failed, this is accessed by another testing tool later in the chain.

**Comment by janekdb** [comment]
Okay, the alternative options are,

1. Cancel PR
2. Add comment to go with this var to aid future explorers
3. Change to `val` and add comment

@klaren Can you help me with these,
- Suggest a comment explaining the purpose of the counter
- Confirm the change to `val` in option 3 will work

2 & 3 will leave the code very slightly improved!

**Comment by klaren** [comment]
@janekdb After some more digging, I found out that we actually don't need the counter at all. I removed the counter instead and is now running CI builds for it. It will be merged when the build is successful.

Thanks for the PR!

**Review by klaren — APPROVED** [review]

**Comment by janekdb** [comment]
@klaren Excellent news! Thanks for your help.

**Pull request closed without merging**